File: 0122_best_time_to_buy_and_sell_stock_II/best_time_stock_II.cpp
#include <iostream>
#include <vector>

using namespace std;

/**
 * Time: O(n)
 * Space: O(1)
 */
int max_profit(vector<int> &prices)
{
    int profits = 0;
    for (int i = 1; i < prices.size(); i++)
        if (prices[i] > prices[i - 1])
            profits += prices[i] - prices[i - 1];
    return profits;
}

File: 0122_best_time_to_buy_and_sell_stock_II/best_time_stock_II.py
from typing import List


def max_profit(prices: List[int]) -> int:
    """
    Time: O(n)
    Space: O(1)
    """
    profits = 0
    for i in range(1, len(prices)):
        if prices[i] > prices[i - 1]:
            profits += prices[i] - prices[i - 1]
    return profits
